File: app/src/main/java/htw/gma_sose22/metronomprokit/AudioGeneratorInterface.java
package htw.gma_sose22.metronomprokit;

public interface AudioGeneratorInterface {

    double[] getSineWave(int samples, int sampleRate, double frequencyOfTone);

    byte[] get16BitPcm(double[] samples);

    void createPlayer();

    void writeSound(double[] samples);

    void destroyAudioTrack();

}

File: app/src/main/java/htw/gma_sose22/metronomprokit/AudioGeneratorInterface.kt
package htw.gma_sose22.metronomprokit

interface AudioGeneratorInterface {
    fun getSineWave(samples: Int, sampleRate: Int, frequencyOfTone: Double): DoubleArray
    fun get16BitPcm(samples: DoubleArray): ByteArray
    fun createPlayer()
    fun writeSound(samples: DoubleArray)
    fun destroyAudioTrack()
}